src/components/CreateAuthor.jsx

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable jsx-a11y/label-has-associated-control */
+import React, { useState } from 'react';
+import { Deserializer } from 'jsonapi-serializer';
+import useAuth from '../hooks/useAuth';
+
+const initialValues = {
+  firstName: '',
+  lastName: '',
+  birthDate: '',
+};
+
+export default function CreateAuthor({ setAuthors }) {
+  const [values, setValues] = useState(initialValues);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const { currentUser } = useAuth();
+
+  const handleSubmit = async function handleSubmit(event) {
+    event.preventDefault();
+    const requestOptions = {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+        Authorization: `Bearer ${currentUser?.access_token}`,
+      },
+      body: JSON.stringify({ ...values }),
+    };
+    fetch(`${process.env.REACT_APP_API_URL}/api/authors`, requestOptions)
+      .then((response) => {
+        if (!response.ok) {
+          setError(true);
+          return {};
+        }
+        return response.json();
+      })
+      .then((data) => {
+        new Deserializer({ keyForAttribute: 'camelCase' })
+          .deserialize(data, (_error, author) => setAuthors((prevState) => [...prevState, author]));
+      })
+      .catch(() => {
+        setError(true);
+        setValues(initialValues);
+      })
+      .finally(() => setLoading(false));
+  };
+
+  const handleChange = function handleChange(event) {
+    setValues((prevValues) => ({
+      ...prevValues,
+      [event.target.name]: event.target.value,
+    }));
+  };
+
+  let isDisabled = true;
+  if (values.firstName && values.lastName && values.birthDate) {
+    isDisabled = false;
+  }
```

Comment:
Una sugerencia para aprovechar React al máximo 🤓
```suggestion
  const isDisabled = useMemo(
    () => !Boolean(values.firstName && values.lastName && values.birthDate),
    [values],
  );
```
Habría que importar `useMemo` de `react` y asegurarse de que ESLint acepta ese array de dependencias, o tiene que ser agregarse cada property de `values`

Reply:
Que buena! Conocía `memo` pero nunca había utilizado `useMemo`, por lo que no lo entendía bien.